src/middlewares/headers-middleware.ts

```diff
@@ -0,0 +1,42 @@
+import { inject, injectable } from 'inversify';
+import { HeaderConfig } from '../common/injectable-interfaces';
+import { isRepositoryEntity } from '../dal/entities/repository-entity';
+import { HeadersConfiguration } from '../http/request/polaris-request-headers';
+import { POLARIS_TYPES } from '../inversion-of-control/polaris-types';
+import { DataVersionFilter } from './middleware-activation-condition/filter-data-version';
+import { RealityIdFilter } from './middleware-activation-condition/filter-realities';
+import {
+    PolarisMiddleware,
+    RequestMiddlewareParams,
+    ResponseMiddlewareParams,
+} from './polaris-middleware';
+
+@injectable()
+export class HeadersMiddleware implements PolarisMiddleware {
```

Comment:
Change name, because this middleware is not about headers
We need to think about better name